samza-operator/src/main/java/org/apache/samza/operators/impl/RootOperatorImpl.java

```diff
@@ -18,19 +18,18 @@
  */
 package org.apache.samza.operators.impl;
 
-import org.apache.samza.operators.data.Message;
+import org.apache.samza.operators.data.IncomingSystemMessage;
 import org.apache.samza.task.MessageCollector;
 import org.apache.samza.task.TaskCoordinator;
 
 
 /**
- * A no-op operator implementation that simply forwards the incoming message to all of its subscribers.
- * @param <M> type of the messages in the input stream.
+ * A no-op operator implementation that forwards the {@link IncomingSystemMessage} to all of its subscribers.
  */
-final class NoOpOperatorImpl<M extends Message> extends OperatorImpl<M, M> {
+final class RootOperatorImpl extends OperatorImpl<IncomingSystemMessage, IncomingSystemMessage> {
```

Comment:
Can we make RootOperatorImpl also a generic-typed class? i.e. RootOperatorImpl<M extends Message>? The reason is to make the entrance stream to a operator DAG flexible s.t. it would be easier to convert a DAG from receiving from system streams to a in-memory MessageStream object. It does not add too much complexity now and would help enable the "dynamic DAG reconfiguration" feature easier in the future.